src/components/sidenav/sidenav.js

```diff
@@ -257,9 +266,36 @@ function SidenavDirective($mdMedia, $mdUtil, $mdConstant, $mdTheming, $animate,
     scope.$watch('isOpen', updateIsOpen);
 
 
+
     // Publish special accessor for the Controller instance
     sidenavCtrl.$toggleOpen = toggleOpen;
 
+    function revalidateVisibility() {
+      var lastValue = scope.isOpen;
+      if (attr.mdIsLockedOpen) return;
+
+      if (isHidden(element, true) && scope.isOpen) {
+        scope.isOpen = false;
+      } else if (!isHidden(element, true) && !scope.isOpen) {
+        scope.isOpen = true;
+      }
+
+      // If the revalidated isOpen Value got changed,
+      // we should apply it to the view without running the updateOpen watcher
+      if (lastValue != scope.isOpen) {
+        skipSidenav = true;
+        if (!scope.$$phase) scope.$apply();
+        skipSidenav = false;
```

Comment:
That won't work because I want to set the `skipSidenav` after running `$apply`, but the normal $timeout `invokeApply` will run the `$apply` after the given function.

See here $timeout snippet
```
        try {
          deferred.resolve(fn.apply(null, args));
        } catch (e) {
          deferred.reject(e);
          $exceptionHandler(e);
        }
        finally {
          delete deferreds[promise.$$timeoutId];
        }

        if (!skipApply) $rootScope.$apply();
```